File: src/my_robot_tutorials/scripts/compute_disk_area_server.py
#!/usr/bin/env python3

from math import pi as M_PI
import rospy
from my_robot_msgs.srv import ComputeDiskArea


def handle_compute_disk_area(req):
    result = M_PI * req.radius ** 2
    rospy.loginfo("Area of circle with radius " + str(req.radius) + " = " + str(result))
    return result


if __name__ == "__main__":
    rospy.init_node("compute_disk_area_server")
    rospy.loginfo("Compute disk area server node created")

    service = rospy.Service(
        "/computer_disk_area",  # Name of service url (address)
        ComputeDiskArea,  # Type of request
        handle_compute_disk_area  # Handle function
    )
    rospy.loginfo("Service Server has been created")

    # Spin to keep the server alive and waiting for requests
    rospy.spin()

File: src/my_robot_tutorials/src/compute_disk_area_server.cpp
#include <cmath>
#include <ros/ros.h>
#include <my_robot_msgs/ComputeDiskArea.h>


// The callback of a server must return a boolean. True: Service Succesful
bool handle_compute_disk_area(
    my_robot_msgs::ComputeDiskArea::Request &req,
    my_robot_msgs::ComputeDiskArea::Response &res
)
{
    float result = M_PI * req.radius * req.radius;
    ROS_INFO("The area of a circle with a radius %f is %f", (float)req.radius, (float)result);
    res.area = result;
    return true;
}


int main(int argc, char **argv)
{
    ros::init(argc, argv, "compute_disk_area_server");
    ros::NodeHandle nh;

    ros::ServiceServer server = nh.advertiseService(
        "/computer_disk_area",
        handle_compute_disk_area
    );

    ros::spin();
}